bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -2439,6 +2439,14 @@ void RewriteInstance::readDynamicRelocations(const SectionRef &Section,
     if (Symbol)
       SymbolIndex[Symbol] = getRelocationSymbol(InputFile, Rel);
 
+    const uint64_t SymAddress = SymbolAddress + Addend;
+    BinaryFunction *Func = BC->getBinaryFunctionContainingAddress(SymAddress);
+    if(Func){
```

Comment:
```suggestion
    if(Func && !(Func->isInConstantIsland(SymAddress)){
```

I think we must check here that the referenced address is not in constant island part of a function